### PR TITLE
Add new bind sample /8

### LINKED ISSUE #30880: An options class , when used as the generic parameter for the `Get<>` method
From [this comment](https://github.com/dotnet/AspNetCore.Docs/pull/30842#pullrequestreview-1702713664_)

> It might be worth adding 'An options class _, when used as the generic parameter for the `Get<>` method_' ...
> 
> ... as `Bind` takes anything, including a concretion of an abstract class, e.g. given:
> 
> ```cs
> public abstract class SomethingWithAName
> {
>     public abstract string Name { get; set; }
> }
> 
> public class PositionOptions : SomethingWithAName
> {
>     public PositionOptions(int age)
>     {
>         Age = age;
>     }
> 
>     public const string Position = "Position";
> 
>     public override string Name { get; set; }
>     public string Title { get; set; } = String.Empty;
> 
>     public int Age { get; set; }
> }
> ```
> 
> ... you can call, and successfully bind with:
> 
> ```cs
> SomethingWithAName p = new PositionOptions(42);
> app.Configuration.Bind("Position", p);
> ```

... you can call, and successfully bind with:

```csharp
SomethingWithAName p = new PositionOptions(42);
app.Configuration.Bind("Position", p);
```

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-7.0

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/configuration/options.md

### Document ID

dc7e3449-650b-d117-5eac-69e5ccb250cb

### Article author

Rick-Anderson

---
[Associated WorkItem - 176941](https://dev.azure.com/msft-skilling/Content/_workitems/edit/176941)

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #30880

### CODE REVIEW

File: aspnetcore/fundamentals/configuration/index/samples/8.x/ConfigSample/Options/NameTitleOptions.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+public class NameTitleOptions(int age) : SomethingWithAName
+{
+    public const string NameTitle = "NameTitle";
+
+    public override string? Name { get; set; }
+    public string Title { get; set; } = string.Empty;
+
+    public int Age { get; set; } = age;
+}
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/configuration/index/samples/8.x/ConfigSample/Options/NameTitleOptions.cs:17]: @SteveDunn your code From [this comment](https://github.com/dotnet/AspNetCore.Docs/pull/30842#pullrequestreview-1702713664)

File: aspnetcore/includes/bind7.md [hunk 2]
```diff
@@ -33,6 +33,14 @@ In the preceding code, by default, changes to the JSON configuration file after
 
 In the preceding code, by default, changes to the JSON configuration file after the app has started are read.
 
+An options class can be used with a generic parameter for the `get<>` method because bind allows generics and the concretion of an abstract class. Consider the following code which uses the abstract class `SomethingWithAName`:
+
+[!code-csharp[](~/fundamentals/configuration/index/samples/8.x/ConfigSample/Options/NameTitleOptions.cs?name=snippet)]
+
+The following code displays the `NameTitleOptions` configuration values:
+
```
Comment (Rick-Anderson) [inline aspnetcore/includes/bind7.md:41]: @SteveDunn please suggest better wording.

**Review by Rick-Anderson — COMMENTED** [review]
@SteveDunn you can ignore most of the changes. GitHub diff does a terrible job where I use the file-scoped namespace declaration. Also fixed a couple null warnings you can ignore.

See [Bind hierarchical configuration](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-8.0#bind-hierarchical-configuration) with changes of the last PR in case you want to make suggestion.

File: aspnetcore/includes/bind7.md (hunk 2 above)
Comment (SteveDunn) [reply aspnetcore/includes/bind7.md]: How about:

> Calls to `Bind` are less strict than calls to `Get<>`; `Bind` allows the concretion of an abstract, but `Get<>` has to create an instance itself...

**Comment by Rick-Anderson** [comment]
@SteveDunn  [Bind hierarchical configuration](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-8.0#bind-hierarchical-configuration) live in 10 minutes. Open an issue and hopefully PR anything you don't like.

**Comment by SteveDunn** [comment]
> @SteveDunn [Bind hierarchical configuration](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-8.0#bind-hierarchical-configuration) live in 10 minutes. Open an issue and hopefully PR anything you don't like.

Thanks @Rick-Anderson !

**Comment by SteveDunn** [comment]
All looks good Rick - great work!

**Pull request merged**